modules/friendlytag.js

```diff
@@ -88,6 +88,28 @@ Twinkle.tag.callback = function friendlytagCallback() {
 		case 'article':
 			Window.setTitle('Article maintenance tagging');
 
+
+			// Build lookups for different sorts: flatObject always
+			// needed, the alphabetical list only for alphabetical display
+			Twinkle.tag.article.flatObject = {};
+			Twinkle.tag.article.alphabeticalList = [];
+			Object.keys(Twinkle.tag.article.tagList).forEach(function(group) {
+				Object.keys(Twinkle.tag.article.tagList[group]).forEach(function(subgroup) {
+					if (Array.isArray(Twinkle.tag.article.tagList[group][subgroup])) {
+						Twinkle.tag.article.tagList[group][subgroup].forEach(function(item) {
+							Twinkle.tag.article.alphabeticalList.push(item.tag);
+							Twinkle.tag.article.flatObject[item.tag] = item.description;
+						});
+					} else {
+						Twinkle.tag.article.alphabeticalList.push(Twinkle.tag.article.tagList[group][subgroup].tag);
+						Twinkle.tag.article.flatObject[Twinkle.tag.article.tagList[group][subgroup].tag] = Twinkle.tag.article.tagList[group][subgroup].description;
+					}
+				});
+			});
+			Twinkle.tag.article.alphabeticalList.sort();
```

Comment:
Duh!

Reply:
Also can you add a comment here noting why we aren't using `Object.values` (which makes things a _lot_ clearer)? Definitely one of first things to fix up when support for IE11 is finally dropped.

Reply:
I was trying to avoid re`sort`ing every time; I doubt folks are switching sort methods very often, but sorting >80 items is nontrivial.  Better still is doing

```js
Twinkle.tag.article.alphabeticalList = Twinkle.tag.article.alphabeticalList || Object.keys(Twinkle.tag.article.flatObject).sort();
```

before the alphabetical generation.  Stores it, but only if and when they actually select the option.  Win-win!